api/worker.go

```diff
@@ -197,32 +229,29 @@ func GetWorker(c *gin.Context) {
 // create a worker in the configured backend.
 func UpdateWorker(c *gin.Context) {
 	// capture middleware values
-	worker := c.Param("worker")
+	u := user.Retrieve(c)
+	w := worker.Retrieve(c)
 
-	logrus.Infof("Updating worker %s", worker)
+	// update engine logger with API metadata
+	//
+	// https://pkg.go.dev/github.com/sirupsen/logrus?tab=doc#Entry.WithFields
+	logrus.WithFields(logrus.Fields{
+		"user":   u.GetName(),
+		"worker": w.GetHostname(),
+	}).Infof("updating worker %s", w.GetHostname())
 
 	// capture body from API request
 	input := new(library.Worker)
 
 	err := c.Bind(input)
 	if err != nil {
-		retErr := fmt.Errorf("unable to decode JSON for worker %s: %w", worker, err)
+		retErr := fmt.Errorf("unable to decode JSON for worker %s: %w", w.GetHostname(), err)
 
 		util.HandleError(c, http.StatusBadRequest, retErr)
 
 		return
 	}
 
-	// send API call to capture the worker
-	w, err := database.FromContext(c).GetWorker(worker)
-	if err != nil {
-		retErr := fmt.Errorf("unable to get worker %s: %w", worker, err)
-
-		util.HandleError(c, http.StatusNotFound, retErr)
-
-		return
-	}
-
```

Comment:
Should this have been removed? Feels not great that the `err = database.FromContext(c).UpdateWorker(w)` call will now return a `http.StatusInternalServerError` if the worker does not exist.

Reply:
> Should this have been removed?

I removed this code in favor of using [the `worker.Retrieve()` middleware function](https://github.com/go-vela/server/blob/master/router/middleware/worker/worker.go#L20-L23):

https://github.com/go-vela/server/blob/0205c1ec656f0fc48c5b6cef87415374f8fa668c/api/worker.go#L228-L233

It felt duplicative to leave it because the endpoint already uses [the `worker.Establish()` middleware function](https://github.com/go-vela/server/blob/master/router/middleware/worker/worker.go#L25-L46):

https://github.com/go-vela/server/blob/0205c1ec656f0fc48c5b6cef87415374f8fa668c/router/worker.go#L34

> Feels not great that the `err = database.FromContext(c).UpdateWorker(w)` call will now return a `http.StatusInternalServerError` if the worker does not exist.

I think we're ok here because `worker.Establish()` returns `404 Not Found` if the worker doesn't exist:

https://github.com/go-vela/server/blob/0863d54c5bd726d80e4655e6d15657882ce6ec29/router/middleware/worker/worker.go#L36-L41

AFAIK, there shouldn't be a change in behavior so I believe we're just re-using existing middleware functions.